File: speakeasy/4.js
#!/usr/bin/env node

process.on('exit', function(code){console.log('EXIT <%s>',code)})
process.on('uncaughtException', function(err){console.log('EXIT uncaughtException <%s>',err); process.exit(1)})

console.log("===Start")

speakeasy=require('speakeasy')
assert = require('assert')
util = require('util')

secret='KRUGS4ZANFZSA2LOORSXEZLTORUW4ZZAK5UGC5BANFZSAZ3PNFXGOIDPNYQE4ZLFMQQGCIDMN5XGOIDTMVXHIZLOMNSSAY3BOJZHSIDPNYQGC3TEEB2GC23FEBWW64TF'
c1=speakeasy.totp({key: secret, encoding : 'base32'});
c2=speakeasy.totp({key: secret, encoding : 'base32'});
console.log("totp with default step=30 encoding in base32, c1 <%s>, c2 <%s>",c1, c2)
assert.equal(c1,c2)

console.log("===Completed successfully!")

/**
ref
1) node, 
https://github.com/markbao/speakeasy/blob/master/test/test_totp.js

2) python
https://pypi.python.org/pypi/onetimepass/0.1.2
**/

/**
seapkeasy api

https://github.com/markbao/speakeasy/blob/master/lib/speakeasy.js

// speakeasy.hotp(options)
//
// Calculates the one-time password given the key and a counter.
//
// options.key                  the key
//        .counter              moving factor
//        .length(=6)           length of the one-time password (default 6)
//        .encoding(='ascii')   key encoding (ascii, hex, or base32)
**/

File: speakeasy/4.py
#!/usr/bin/env python


import base64
my_key="This is interesting What is going on Need a long sentence carry on and take more"
secret = base64.b32encode(my_key)
print secret

from otpauth import OtpAuth

def otpauth_totp(f_key):
  print "\n1.b test totp"
  auth = OtpAuth(f_key) # default step=30
  code = auth.totp()
  assert auth.valid_totp(code)
  print "secret <%s> code <%s>" % (auth.secret, code)
  #print dir(auth) 
  print "completed successfully"

otpauth_totp(my_key)



'''
A) Algorithm recap:

1 - in python, choose a KEY
2 - in python, encode key in base32 as SECRET, to be used in speakeasy in node
3 - in python, generate token to totp using KEY 

4 - in  node,  generate token using base32 SECRET

5 - toke in 3 and token on 4 are the same

B) ref:
https://github.com/lepture/otpauth/blob/master/otpauth.py

class OtpAuth(object):
    """One Time Password Authentication.

  :param secret: A secret token for the authentication.
    """

    def __init__(self, secret):

    def totp(self, period=30):
'''
